sim_pipeline/sim_lsst_galaxy_galaxy_lens_simulation.py

```diff
@@ -0,0 +1,283 @@
+        :param galaxy_type: 'blue' or 'red' (source or lens)
+        :param position_scatter:
+        """
+        self.pipeline = pipeline
+        self.mag_cut = mag_cut
+        self.z_min = z_min
+        self.z_max = z_max
+        self.galaxy_type = galaxy_type
+        self.position_scatter = position_scatter
+        self.cosmo = cosmo
+
+    def get_source(self):
+        """
+        Define the source cut based on redshift, magnitude and galaxy type
+        return the information
+        """
+        source_cut = (self.pipeline[self.galaxy_type]['z'] > self.z_min) & (
+                    self.pipeline[self.galaxy_type]['z'] < self.z_max) & (
+                                 self.pipeline[self.galaxy_type]['mag_g'] < self.mag_cut)
+        pipeline_cut = self.pipeline[self.galaxy_type][source_cut]
+        n = len(pipeline_cut)
+        index = random.randint(0, n - 1)
+        source = pipeline_cut[index]
+        return source
+
+    def get_params(self):
+        """
+        Method to get the galaxy parameters to be used in the simulation
+        return the band information g,r,i; (including the magnitude, size, shape, and position under random scatter)
+               the distance between the galaxy and observer;
+               the redshift of the galaxy;
+               the stellar mass; (for velocity dispersion)
+
+        """
+        source= self.get_source()
+        stellar_mass = source['stellar_mass']
+        redz = source['z']
+        distan = [self.cosmo.angular_diameter_distance(redz).value]    # distance between galaxy and observer
+        size_arcsec = source['angular_size'] / constants.arcsec   # convert radian to arc seconds
+        mag_g, mag_r, mag_i = source['mag_g'], source['mag_r'], source['mag_i']
+        mag_apparent = source['M']
+        e = source['ellipticity']
+        phi = np.random.uniform(0, np.pi)
+        e1 = e * np.cos(phi)
+        e2 = e * np.sin(phi)
+        if self.galaxy_type == 'blue':
+            n_sersic = 1
+        else:
+            n_sersic = 4
+        center_x, center_y = np.random.uniform(-self.position_scatter, self.position_scatter), np.random.uniform(
+            -self.position_scatter, self.position_scatter)
+        kwargs_galaxy_g = [{'magnitude': mag_g, 'R_sersic': size_arcsec, 'n_sersic': n_sersic,
+                            'e1': e1, 'e2': e2, 'center_x': center_x, 'center_y': center_y}]
+        kwargs_galaxy_r = [{'magnitude': mag_r, 'R_sersic': size_arcsec, 'n_sersic': n_sersic,
+                            'e1': e1, 'e2': e2, 'center_x': center_x, 'center_y': center_y}]
+        kwargs_galaxy_i = [{'magnitude': mag_i, 'R_sersic': size_arcsec, 'n_sersic': n_sersic,
+                            'e1': e1, 'e2': e2, 'center_x': center_x, 'center_y': center_y}]
+        return [kwargs_galaxy_g, kwargs_galaxy_r, kwargs_galaxy_i], distan, source['z'], stellar_mass
+
+def get_vd_from_stallermass(smass):
```

Comment:
I would put this function in a different file that is tasked to link stellar mass and halo masses